WaterLog_Backend/Controllers/SegmentLeaksController.cs

```diff
@@ -61,24 +61,28 @@ public async Task<ActionResult<string>> GetLitres(int id)
 
         //Resolve Leakage
         [HttpPut("resolve/{id}")]
-        public async Task<IActionResult> Resolve(int id)
+        public async Task<ActionResult<IEnumerable<SegmentLeaksEntry>>> Resolve(int id)
         {
-            var leaks = await _db.SegmentLeaks.FindAsync(id);
+            var leaks = await _db.SegmentLeaks.Where(s => s.SegmentsId == id).FirstOrDefaultAsync();
             if (leaks == null)
             {
                 return NotFound();
             }
-            leaks.ResolvedStatus = "resolved";
-            _db.SegmentLeaks.Update(leaks);
+            if (leaks.ResolvedStatus == "unresolved")
+            {
+                leaks.ResolvedStatus = "resolved";
 
-            // post to Historylogs
-            var hist = new HistoryLogEntry();
-            hist.Date = leaks.LatestTimeStamp;
-            hist.EventsId = leaks.Id;
-            hist.Type = EnumTypeOfEvent.LEAK;
-            await _db.HistoryLogs.AddAsync(hist);
-            await _db.SaveChangesAsync();
-            return Ok();
+                var hist = new HistoryLogEntry();
+                hist.Date = DateTime.Now;
+                hist.EventsId = leaks.SegmentsId;
+                hist.Type = EnumTypeOfEvent.LEAK;
```

Comment:
this can also be done like 
`var hist = new History{
hist.Date = DateTime.Now,
hist.EventsId = leaks.SegmentsId,
hist.Type = EnumTypeOfEvent.LEAK
}`

Reply:
But on separate lines, for some reason this text editor is being weird. However Its just suggestion. you dont have to do it this way if u dont want to

Reply:
I tried making the changes but visual studio mourns, otherwise also don't understand the change, is it having the assignments in one line?

Reply:
Nah its cool remind me on Friday will show you what I meant

Reply:
Ok thanks, will talk on Friday